pkg/informer/informer.go

```diff
@@ -219,15 +249,115 @@ func (d *scopedDynamicDiscoverySharedInformerFactory) ForResource(gvr schema.Gro
 	return clusterInformer.Cluster(d.cluster), nil
 }
 
+// NewSingleClusterDynamicDiscoverySharedInformerFactory returns a factory for
+// cluster-unaware (standard Kube) shared informers that discovers new types
+// and informs on updates to resources of those types.
+// It retrieves GVR-related information from CRD informers.
+func NewSingleClusterDynamicDiscoverySharedInformerFactory(
+	cfg *rest.Config,
+	filterFunc func(obj interface{}) bool,
+	crdInformer kcpapiextensionsv1informers.CustomResourceDefinitionClusterInformer,
+	indexers cache.Indexers,
+) (*SingleClusterDynamicDiscoverySharedInformerFactory, error) {
+	crdIndexer := crdInformer.Informer().GetIndexer()
+
+	// Add an index function that indexes a CRD by its group/version/resource.
+	if err := crdInformer.Informer().AddIndexers(cache.Indexers{
+		byGroupVersionResourceIndex: byGroupVersionResourceIndexFunc,
+	}); err != nil {
+		return nil, err
+	}
+
+	return NewSingleClusterDelegatingDynamicDiscoverySharedInformerFactory(cfg, filterFunc, indexers, &crdGVRSource{
+		crdInformer: crdInformer,
+		crdIndexer:  crdIndexer,
+	})
+}
+
+// NewSingleClusterDynamicDiscoverySharedInformerFactory returns a factory for
+// cluster-unaware (standard Kube) shared informers that discovers new types
+// and informs on updates to resources of those types.
+// It receives the GVR-related information by delegating to a GVRSource.
+func NewSingleClusterDelegatingDynamicDiscoverySharedInformerFactory(
```

Comment:
Why don't we just have one parameterized constructor? What's the diff here?

Reply:
I added a parameterized constructor with the common code.